src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,349 @@
+            #  pseudos should not change
+            #  what about calc_type, restart_from_deps and prev_outputs_deps ?
+            if param in [
+                "calc_type",
+                "pseudos",
+                "extra_abivars",
+                "restart_from_deps",
+                "prev_outputs_deps",
+            ]:
+                continue
+            if param in params:
+                input_set_generator.__setattr__(param, params.pop(param))
+            elif hasattr(prev_gen, param):
+                input_set_generator.__setattr__(param, prev_gen.__getattribute__(param))
+        name = name or cls.name
+        maker = cls(
+            input_set_generator=input_set_generator, name=name, wall_time=wall_time
+        )
+        return maker
+
+    def __post_init__(self):
+        """Process post-init configuration."""
+        self.critical_events = [
+            as_event_class(ce_name) for ce_name in self.CRITICAL_EVENTS
+        ]
+        if self.input_set_generator is None:
+            raise InitializationError("Input set generator is not set.")
+
+    @property
+    def calc_type(self):
+        return self.input_set_generator.calc_type
+
+    @job
+    def make(
+        self,
+        structure: Optional[Structure] = None,
+        prev_outputs: Optional[Any] = None,
+        restart_from: Optional[Union[str]] = None,
+        history: Optional[JobHistory] = None,
+    ) -> Union[jobflow.Flow, jobflow.Job]:
+        """
+        Return an ABINIT jobflow.Job.
+
+        Parameters
+        ----------
+        structure : Structure
+            A pymatgen structure object.
+        prev_outputs : TODO: add description from sets.base
+        restart_from : TODO: add description from sets.base
+        history : JobHistory
+            A JobHistory object containing the history of this job.
+        """
+        # Setup job and get general job configuration
+        config = setup_job(
+            structure=structure,
+            prev_outputs=prev_outputs,
+            restart_from=restart_from,
+            history=history,
+            wall_time=self.wall_time,
+        )
+
+        # Write abinit input set
+        write_abinit_input_set(
+            structure=structure,
+            input_set_generator=self.input_set_generator,
+            prev_outputs=prev_outputs,
+            restart_from=restart_from,
+            directory=config.workdir,
+        )
+
+        # Run abinit
+        run_status = run_abinit(
+            abinit_cmd=SETTINGS.ABINIT_CMD,
+            mpirun_cmd=SETTINGS.ABINIT_MPIRUN_CMD,
+            wall_time=config.wall_time,
+            start_time=config.start_time,
+        )
+
+        # parse Abinit outputs
+        run_number = config.history.run_number
+        task_doc = AbinitTaskDocument.from_directory(
+            config.workdir,
+            critical_events=self.critical_events,
+            run_number=run_number,
+            run_status=run_status,
```

Comment:
For example, there might later on be custom settings in the `AbinitTaskDocument` e.g., for parsing specific objects or running a certain analysis.

You could then have a `task_document_kwargs` field on the base abinit maker that could be used to control these options.